core/pkg/evaluator/json.go

```diff
@@ -400,16 +400,36 @@ func getFlagdProperties(context map[string]any) (flagdProperties, bool) {
 	return p, true
 }
 
+func (je *JSON) loadAndCompileSchema() *gojsonschema.Schema {
+	schemaLoader := gojsonschema.NewSchemaLoader()
+
+	targetingSchemaLoader := gojsonschema.NewStringLoader(schema.Targeting)
+	if err := schemaLoader.AddSchemas(targetingSchemaLoader); err != nil {
+		je.Logger.Warn(fmt.Sprintf("error adding Targeting schema: %s", err))
+	}
+
+	flagdDefinitionsLoader := gojsonschema.NewStringLoader(schema.FlagdDefinitions)
+	compiledSchema, err := schemaLoader.Compile(flagdDefinitionsLoader)
+	if err != nil {
+		je.Logger.Warn(fmt.Sprintf("error compiling FlagdDefinitions schema: %s", err))
+	}
+
+	return compiledSchema
+}
+
 // configToFlags convert string configurations to flags and store them to pointer newFlags
 func (je *JSON) configToFlags(config string, newFlags *Flags) error {
-	schemaLoader := gojsonschema.NewStringLoader(schema.FlagdDefinitions)
+	compiledSchema := je.loadAndCompileSchema()
+
 	flagStringLoader := gojsonschema.NewStringLoader(config)
 
-	result, err := gojsonschema.Validate(schemaLoader, flagStringLoader)
+	result, err := compiledSchema.Validate(flagStringLoader)
 	if err != nil {
-		return fmt.Errorf("error validating json schema: %w", err)
+		je.Logger.Warn(fmt.Sprintf("failed to execute JSON schema validation: %s", err))
 	} else if !result.Valid() {
-		return fmt.Errorf("JSON schema validation failed: %s", buildErrorString(result.Errors()))
+		je.Logger.Warn(fmt.Sprintf(
+			"JSON data does not conform to the schema. Validation errors: %s", buildErrorString(result.Errors()),
```

Comment:
```suggestion
			"flag definition does not conform to the schema; validation errors: %s", buildErrorString(result.Errors()),
```

Let's make it very clear to people the validation that failed is for the flag definitions.